Developer-instructions.md

```diff
@@ -0,0 +1,260 @@
+# Developer instructions
+
+## Recommended setup
+
+Our recommended setup for working on the library is by using vscode on Linux, MacOS, or WSL, with
+the coq-lsp and ocaml extensions. We will now describe more extensively how to achieve this setup.
+
+### Installation of opam
+
+A more extensive description can be found [here](https://ocaml.org/docs/installing-ocaml).
+This is a short description.
+
+On some versions of Linux and WSL, you can install opam with
+```
+apt-get install opam
+```
+On MacOs, you can either use homebrew
+```
+brew install gpatch
+brew install opam
+```
+or macports
+```
+port install opam
+```
+
+### Creating the desired opam environment: simplified version
+When we develop, we need to be aware of the version of Coq we develop for.
+Most development is done based off the main branch, and then the following setup will suffice.
+
+```
+opam init
+eval $(opam env)
+opam install coq-lsp.2.0.0+8.17
+opam install ocaml-lsp-server
+```
+replacing 8.17 with the desired version of Coq, or more generally, replacing 
+2.0.0+8.17 by the desired version of coq-lsp.
+
+### Creating the desired opam environment: advanced version for supporting multiple versions of Coq
+
+When we bring changes into different versions of Coq (for instance for the branches 8.18, 8.19...)
+it is convenient to have multiple opam environement, i.e. switches, available. If you prefer to start
+with such a setup anyway, you can run
+```
+opam init --bare
+``` 
+Regardless of how you initialized opam, you can now add a new switch with
+```
+opam switch create your_preferred_switch_name ocaml-base-compiler.4.14.1
+```
+Next you can install the background libraries again
+```
+opam install coq-lsp.2.0.0+8.17
```

Comment:
Think you folks got a typo here, do you mean `coq-lsp.0.2.0+8.17` ?

Reply:
Oops! You're right, thanks!